scripts/pi-hole/php/groups.php

```diff
@@ -877,12 +877,29 @@ function verify_ID_array($arr)
     try {
         $db->query('BEGIN TRANSACTION;');
 
+        $domain = html_entity_decode(trim($_POST['domain']));
+        // Convert domain name to IDNA ASCII form for international domains
+        if (extension_loaded("intl")) {
+            // Be prepared that this may fail and see our comments above
+            // (search for "idn_to_utf8)
+            $idn_domain = false;
+            if (defined("INTL_IDNA_VARIANT_UTS46")) {
+                $idn_domain = idn_to_ascii($domain, IDNA_NONTRANSITIONAL_TO_ASCII, INTL_IDNA_VARIANT_UTS46);
+            }
+            if ($idn_domain === false && defined("INTL_IDNA_VARIANT_2003")) {
+                $idn_domain = idn_to_ascii($domain, IDNA_DEFAULT, INTL_IDNA_VARIANT_2003);
+            }
+            if($idn_domain !== false) {
+                $domain = $idn_domain;
+            }
```

Comment:
No need to use the extra variable `$idn_domain`:

```suggestion
            if (defined("INTL_IDNA_VARIANT_UTS46")) {
                $domain = idn_to_ascii($domain, IDNA_NONTRANSITIONAL_TO_ASCII, INTL_IDNA_VARIANT_UTS46);
            } elseif (defined("INTL_IDNA_VARIANT_2003")) {
                $domain = idn_to_ascii($domain, IDNA_DEFAULT, INTL_IDNA_VARIANT_2003);
            }
```

Reply:
Thanks :-)